plenum/server/catchup/seeder_service.py

```diff
@@ -2,27 +2,60 @@
 
 from plenum.common.channel import RxChannel
 from plenum.common.ledger import Ledger
-from plenum.common.messages.node_messages import CatchupReq, CatchupRep, ConsistencyProof
+from plenum.common.messages.node_messages import CatchupReq, CatchupRep, ConsistencyProof, LedgerStatus
 from plenum.common.util import SortedDict
-from plenum.server.catchup.utils import CatchupDataProvider
+from plenum.server.catchup.utils import CatchupDataProvider, build_ledger_status
 from stp_core.common.log import getlogger
 
 logger = getlogger()
 
 
 class SeederService:
-    def __init__(self, input: RxChannel, provider: CatchupDataProvider):
+    def __init__(self, input: RxChannel, provider: CatchupDataProvider, service_clients: bool):
```

Comment:
I would avoid using such a boolean in a parent class when we have children corresponding to this flag

Reply:
Fixed